sklearn/cluster/affinity_propagation_.py

```diff
@@ -403,7 +403,7 @@ def predict(self, X):
             raise ValueError("Predict method is not supported when "
                              "affinity='precomputed'.")
 
-        if self.cluster_centers_.size > 0:
+        if self.cluster_centers_.shape[0] > 0:
```

Comment:
in fact there was a bug with the random seed I had the cluster centers were [[0, 0..0]] (unique zero cluster center), and in the sparse version it was the same but in a sparse representation hence doing size on it returned 0

Reply:
This sensitivity to the random seed made me think that maybe I should improve the test by testing several random seeds and only asserting that the result is close to a certain tolerance ?

Reply:
I just opened a PR here #13334 to isolate the change, I'll remove it from this PR